### PR TITLE
[ENG-768] Fix importing providers with brands

### PR DESCRIPTION
**Pull request by Johnetordoff**
## Purpose

Currently our import provider from JSON feature doesn't work if the imported provider has a brand associated with it, this PR fixes that and consolidates the import views to reduce maintenance in the feature.

## Changes

- consolidates import views for absract provider classes.

## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify that you can import json provider files over various environments.

## Documentation

Bug fix, no docs.

## Side Effects

None that I know of.

## Ticket

https://openscience.atlassian.net/browse/ENG-768

### CODE REVIEW

**Review by corbinSanders — APPROVED** [review]
Looks good! Thanks for the cleanup too!

File: osf/models/provider.py [hunk 1]
```diff
@@ -46,6 +48,65 @@ class Meta:
...
+            try:
+                brand = Brand.objects.get(id=provider_data.pop('brand'))
+            except Brand.DoesNotExist:  # JSON is old or imported from staging etc.
+                pass
+
+        if provider_data.get('id'):  # <- determines if creating new or updating
+            provider = cls.objects.get(id=provider_data.pop('id'))
+        else:
+            try:
+                provider = RegistrationProvider(**provider_data)
```
Comment (Johnetordoff) [inline osf/models/provider.py:72]: should be cls

File: osf/models/provider.py [hunk 2]
```diff
@@ -46,6 +48,65 @@ class Meta:
...
+                provider.save()
+            except ValidationError as e:
+                if 'Abstract provider with this  id and Type already exists.' in str(e):
+                    provider_data.pop('_id')  # these must be unique
+                    provider = RegistrationProvider(**provider_data)
```
Comment (Johnetordoff) [inline osf/models/provider.py:77]: should be cls

File: osf/models/provider.py [hunk 3]
```diff
@@ -46,6 +48,65 @@ class Meta:
...
+                pass
+
+        if provider_data.get('id'):  # <- determines if creating new or updating
+            provider = cls.objects.get(id=provider_data.pop('id'))
+        else:
+            try:
+                provider = RegistrationProvider(**provider_data)
+                provider.save()
+            except ValidationError as e:
+                if 'Abstract provider with this  id and Type already exists.' in str(e):
```
Comment (Johnetordoff) [inline osf/models/provider.py:75]: Don't do this! no string checking

File: osf/models/provider.py [hunk 4]
```diff
@@ -46,6 +48,65 @@ class Meta:
...
+            except Brand.DoesNotExist:  # JSON is old or imported from staging etc.
+                pass
+
+        if provider_data.get('id'):  # <- determines if creating new or updating
+            provider = cls.objects.get(id=provider_data.pop('id'))
+        else:
+            try:
+                provider = RegistrationProvider(**provider_data)
+                provider.save()
+            except ValidationError as e:
```
Comment (Johnetordoff) [inline osf/models/provider.py:74]: remember to raise something! if new error.

**Pull request closed without merging**